datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -1508,8 +1547,89 @@ public void handleDataChange(String dataPath, Object data) throws Exception {
     }
 
     @Override
-    public void handleDataDeleted(String dataPath) throws Exception {
+    public void handleDataDeleted(String dataPath) {
       // do nothing
     }
   }
+
+  /**
+   * Listener for ZooKeeper state changes.
+   */
+  private class ZkStateChangeListener implements IZkStateListener {
+    private Timer _timer;
```

Comment:
I think the code would be a lot simpler if we used a `ScheduledExecutorService` instead of this `Timer`. 

Here's what I'm thinking:
- We declare two fields:
    ```java
        private final ScheduledExecutorService _scheduledExecutorService;
        private Future<?> _zkSessionExpiryFuture = CompletableFuture.completedFuture(null);
    ```
- We change `cancelTimer()` to do:
   ```java
   _zkSessionExpiryFuture.cancel(false);
   ```
- We change `scheduleExpiryTimerAfterSessionTimeout()` to do:
    ```java
     _zkSessionExpiryFuture =
          _scheduledExecutorService.schedule(ZkAdapter.this::onSessionExpired, _sessionTimeoutMs,
              TimeUnit.MILLISECONDS);
    ```

Reply:
+1, I like this suggestion more than using the `_timer`

Reply:
Done.